src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
+
+  health = 100;
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden !== true && animal instanceof Herbivore) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.removeDeadAnimals();
```

Comment:
[CODE PERFOMANCE] - Instead of checking for the animal's health every time after biting and calling `removeDeadAnimals`, you could filter out dead animals in a more efficient way. Consider calling `Animal.removeDeadAnimals` after the `bite` method has been executed on all animals, rather than after each individual bite, to reduce the number of times the array is filtered.